minimum_recolors_to_get_k_consecutive_black_blocks.cpp

```diff
@@ -0,0 +1,40 @@
+// Author: Hacer Akıncı
+// Question Link: https://leetcode.com/problems/minimum-recolors-to-get-k-consecutive-black-blocks/
+// Time compexity: O(n) 
+// Space compexity: O(1) 
+
+class Solution {
+public:
+    
+    int minimumRecolors(string blocks, int k) {
+        
+        int whiteBlocks = 0, i;
+        
+        // We calculate white blocks number of the first substring of length k.
+        for(i = 0; i < k; i++){
+            if(blocks[i] == 'W') whiteBlocks++;
+        }
+        
+        // We initalize minOperation with first value of whiteBlocks. Then check rest of the string for minimum value . 
+        int minOperation = whiteBlocks;
+        
+        while(i < blocks.size()){

```

Comment:
Also maybe using a for loop makes this to understand better since boundaries are defined in the for loop